File: prob-1-multiples-of-3-and-5/prob-1-multiples-of-3-and-5.js
var total = 0;

/*for(var i = 0; i < 1000; i++) {
	if((i % 3 == 0) || (i % 5 == 0)) {
		total += i;
	}
}*/

//Optimized Solution

function sumDivisibleBy(n) {
	var p = parseInt(999 / n, 10)
	return parseInt(n * p * (p + 1) / 2, 10)
}

total = sumDivisibleBy(3) + sumDivisibleBy(5) - sumDivisibleBy(15)

console.log(total)

File: prob-1-multiples-of-3-and-5/prob-1-multiples-of-3-and-5.c
#include <stdio.h>

int main() {
	int total = 0;
	//int i = 0;

	/*for(i = 0; i < 1000; i++) {
		if((i % 3 == 0) || (i % 5 == 0)) {
			total += i;
		}
	}*/

	//Optimized Solution
	
	int target = 999;

	int sumDivisibleBy(int n) {
		int p;
		p = target / n;
		return (n * p * (p + 1)) / 2;
	}

	total = sumDivisibleBy(3) + sumDivisibleBy(5) - sumDivisibleBy(15);
	
	printf("%d\n", total);
}
